classes/components/PKPStatsJobsTable.inc.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * @file components/PKPStatsJobsTable.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPStatsJobsTable
+ * @ingroup classes_components_stats
+ *
+ * @brief A class to prepare the data object for a jobs table UI component
+ */
+
+namespace PKP\components;
+
+use APP\i18n\AppLocale;
+
+class PKPStatsJobsTable
+{
+    /** @var string An ID for this component  */
+    public $id = '';
+
+    /** @var array Configuration for the columns to display in the table */
+    public $tableColumns = [];
+
+    /** @var array Rows to display in the table */
+    public $tableRows = [];
+
+    /** @var null|string Table description */
+    public $description = null;
+
+    /** @var null|string Table label */
+    public $label = null;
+
+    /**
+     * Constructor
+     *
+     * @param $args array Optional arguments
+     */
+    public function __construct(string $id, array $args = [])
+    {
+        AppLocale::requireComponents([
+            LOCALE_COMPONENT_PKP_MANAGER,
+            LOCALE_COMPONENT_APP_MANAGER
+        ]);
+
+        $this->id = $id;
+        $this->init($args);
+    }
+
+    /**
+     * Initialize the handler with config parameters
+     *
+     * @param $args array Configuration params
+     */
+    public function init($args = [])
```

Comment:
It'll be nice to use named parameters to replace this kind of pattern, once we have a PHP8.0 baseline! A fluent interface would work here too, but I'm not sure it's worth the extra verbiage.

Reply:
I agree with the named parameters usage, but I don't know about PHP 8.0 adoption across our community base. I think this could be added in a near future.